Products/PloneFormGen/content/fields.py

```diff
@@ -505,20 +507,36 @@ def __init__(self, oid, **kwargs):
         self.fgField = DateTimeField('fg_date_field',
             searchable=0,
             required=0,
+            accessor = 'nullAccessor',
             write_permission = View,
-            widget=CalendarWidget(),
+            widget=DatetimeWidget(),
             )
 
-
     security.declareProtected(ModifyPortalContent, 'setFgShowHM')
     def setFgShowHM(self, value, **kw):
         """ set show_hm """
-
-        if type(value) == BooleanType:
-            self.fgField.widget.show_hm = value
+        if not type(value) == BooleanType:
```

Comment:
This should be `!=` instead of `not ==`. But we should have been using `isinstance` anyway.